testing/android/native_activity/native_activity.cc

```diff
@@ -0,0 +1,142 @@
+                                                  ANativeWindow* window) {
+    reinterpret_cast<NativeActivity*>(activity->instance)
+        ->OnNativeWindowResized(window);
+  };
+  activity->callbacks->onNativeWindowRedrawNeeded =
+      [](ANativeActivity* activity, ANativeWindow* window) {
+        reinterpret_cast<NativeActivity*>(activity->instance)
+            ->OnNativeWindowRedrawNeeded(window);
+      };
+  activity->callbacks->onNativeWindowDestroyed = [](ANativeActivity* activity,
+                                                    ANativeWindow* window) {
+    reinterpret_cast<NativeActivity*>(activity->instance)
+        ->OnNativeWindowDestroyed(window);
+  };
+  activity->callbacks->onInputQueueCreated = [](ANativeActivity* activity,
+                                                AInputQueue* queue) {
+    reinterpret_cast<NativeActivity*>(activity->instance)
+        ->OnInputQueueCreated(queue);
+  };
+  activity->callbacks->onInputQueueDestroyed = [](ANativeActivity* activity,
+                                                  AInputQueue* queue) {
+    reinterpret_cast<NativeActivity*>(activity->instance)
+        ->OnInputQueueDestroyed(queue);
+  };
+  activity->callbacks->onConfigurationChanged = [](ANativeActivity* activity) {
+    reinterpret_cast<NativeActivity*>(activity->instance)
+        ->OnConfigurationChanged();
+  };
+  activity->callbacks->onLowMemory = [](ANativeActivity* activity) {
+    reinterpret_cast<NativeActivity*>(activity->instance)->OnLowMemory();
+  };
+}
+
+NativeActivity::~NativeActivity() = default;
+
+void NativeActivity::OnStart() {}
+
+void NativeActivity::OnStop() {}
+
+void NativeActivity::OnPause() {}
+
+void NativeActivity::OnResume() {}
+
+std::shared_ptr<fml::Mapping> NativeActivity::OnSaveInstanceState() {
+  return nullptr;
+}
+
+void NativeActivity::OnWindowFocusChanged(bool has_focus) {}
+
+void NativeActivity::OnNativeWindowCreated(ANativeWindow* window) {}
+
+void NativeActivity::OnNativeWindowResized(ANativeWindow* window) {}
+
+void NativeActivity::OnNativeWindowRedrawNeeded(ANativeWindow* window) {}
+
+void NativeActivity::OnNativeWindowDestroyed(ANativeWindow* window) {}
+
+void NativeActivity::OnInputQueueCreated(AInputQueue* queue) {}
+
+void NativeActivity::OnInputQueueDestroyed(AInputQueue* queue) {}
+
+void NativeActivity::OnConfigurationChanged() {}
+
+void NativeActivity::OnLowMemory() {}
+
+void NativeActivity::Terminate() {
+  ANativeActivity_finish(activity_);
+}
+
+}  // namespace flutter
+
+extern "C" __attribute__((visibility("default"))) void ANativeActivity_onCreate(
+    ANativeActivity* activity,
+    void* saved_state,
+    size_t saved_state_size) {
+  std::unique_ptr<fml::Mapping> saved_state_mapping;
+  if (saved_state_size > 0u) {
+    saved_state_mapping = std::make_unique<fml::MallocMapping>(
+        fml::MallocMapping::Copy(saved_state, saved_state_size));
+  }
+  flutter::NativeActivityMain(activity, std::move(saved_state_mapping))
```

Comment:
Comment that the `NativeActivity` will be deleted in the activity's `onDestroy` callback

Reply:
Done.